backend/clickhouse/migrations/000082_create_new_traces_by_id_mv.up.sql

```diff
@@ -0,0 +1,60 @@
+CREATE TABLE IF NOT EXISTS traces_by_id_new (
+    `Timestamp` DateTime64(9),
+    `UUID` UUID,
+    `TraceId` String,
+    `SpanId` String,
+    `ParentSpanId` String,
+    `ProjectId` UInt32,
+    `SecureSessionId` String,
+    `TraceState` String,
+    `SpanName` LowCardinality(String),
+    `SpanKind` LowCardinality(String),
+    `Duration` Int64,
+    `ServiceName` LowCardinality(String),
+    `ServiceVersion` String,
+    `TraceAttributes` Map(LowCardinality(String), String),
+    `StatusCode` LowCardinality(String),
+    `StatusMessage` String,
+    `Environment` String,
+    `HasErrors` Bool,
+    `Events.Timestamp` Array(DateTime64(9)),
+    `Events.Name` Array(LowCardinality(String)),
+    `Events.Attributes` Array(Map(LowCardinality(String), String)),
+    `Links.TraceId` Array(String),
+    `Links.SpanId` Array(String),
+    `Links.TraceState` Array(String),
+    `Links.Attributes` Array(Map(LowCardinality(String), String))
+)
+    ENGINE = MergeTree
+        PARTITION BY substring(toString(UUID), 1, 1) 
```

Comment:
Does it make sense to partition by the UUID if then a single trace could have spans across different partitions?

Reply:
I think it should be partitioned by traceId instead. Because traceId can be either a UUID or one of our own requestIds, maybe using something like `farmHash64(TraceId) % 100` would be best? I think we saw that a reduction of about 100x would work well for our own project

Reply:
@mayberryzane I like that, and makes sense based on the trace id! Quick question: this would essentially bring us from 1 -> 100 partitions. Curious if we have a method for deciding how many partitions is the right amount?

Reply:
makes sense why we should use the hash fn. does it make more sense to use TraceId here because we would want to load multiple spans for a given trace id from the same part rather than across parts (which would be the case if we partition by UUID)?

Reply:
That is my understanding, but will let Zane fill in

Resolving to merge